app/controllers/registrant/sessions_controller.rb

```diff
@@ -97,4 +95,18 @@ def find_user_by_idc(idc)
     return User.new unless idc
     ApiUser.find_by(identity_code: idc) || User.new
   end
-end
+
+  private
+
+  def after_sign_in_path_for(_resource_or_scope)
+    registrant_root_path
+  end
+
+  def after_sign_out_path_for(_resource_or_scope)
+    new_registrant_user_session_path
+  end
+
+  def user_for_paper_trail
+    current_registrant_user.present? ? current_registrant_user.id_role_username : 'anonymous'
```

Comment:
AFAIR, RubyMine for whatever reason suggests usage of ternary operators, but this is a dressed-up [Nil-Check](https://github.com/troessner/reek/blob/master/docs/Nil-Check.md). A more `tell, don't ask` way to express the same thing would be:

```ruby
def user_for_paper_trail
  current_registrant_user&.id_role_username || 'anonymous'
end

Reply:
I think `AnonymousUser` class might be even more suitable.

Reply:
In general yes, but for that we'd need bigger refactoring, as there are other places that check if `current_user` is NIl or not:

https://github.com/internetee/registry/blob/master/app/controllers/registrant/domain_delete_confirms_controller.rb#L22
https://github.com/internetee/registry/blob/master/app/views/layouts/registrant/application.html.erb#L40